cnudie/util.py

```diff
@@ -589,3 +595,87 @@ def desanitise_version(version: str) -> str:
     desanitised_version = version.replace(META_SEPARATOR, '+')
 
     return desanitised_version
+
+
+def resource_content(
+    resource_node: 'cnudie.iter.ResourceNode',
+    oci_client: oci.client.Client,
+    s3_client: 'botocore.client.S3',
+) -> collections.abc.Generator[bytes, None, None]:
+    '''
+    returns a generator yielding tarfile stream containing resource contents as members
+    raises if resource or access type is not supported
+    '''
+    component: cm.Component = resource_node.component
+    resource: cm.Resource = resource_node.resource
+
+    if resource.access.type is cm.AccessType.OCI_REGISTRY:
+        if resource.type is not cm.ResourceType.OCI_IMAGE:
+            return
+
+        def iter_content():
+            image_reference = gci.oci.image_ref_with_digest(
+                image_reference=resource.access.imageReference,
+                digest=resource.digest,
+                oci_client=oci_client,
+            )
+            yield from oci.image_layers_as_tarfile_generator(
+                image_reference=image_reference,
+                oci_client=oci_client,
+                include_config_blob=False,
+                fallback_to_first_subimage_if_index=True,
+            )
+
+        return iter_content()
+
+
+    elif resource.access.type is cm.AccessType.S3:
+        if resource.type != 'application/tar+vm-image-rootfs':
+            return
+
+        def as_blob_descriptors():
+            yield cnudie.access.s3_access_as_blob_descriptor(
+                s3_client=s3_client,
+                s3_access=resource.access,
+            )
+
+        return tarutil.concat_blobs_as_tarstream(
+            blobs=as_blob_descriptors(),
+        )
+
+    elif resource.access.type is cm.AccessType.LOCAL_BLOB:
+        img_ref = oci_artefact_reference(
+            component=f'{component.name}:{component.version}',
+            ocm_repository=component.current_repository_ctx(),
+        )
+        digest = resource.access.localReference
+        size = resource.access.size
+
+        if resource.access.globalAccess:
```

Comment:
Since you're overwriting all variables from before, why not putting both into the conditional block?